src/app/features/sidebar/queries-section/index.tsx

```diff
@@ -126,6 +128,22 @@ const LocalQueriesView = ({toolbarButtons}) => {
   const flatQueries = flatLib.filter((n) => !("items" in n))
   const filteredQueriesCount = flattenQueryTree(filteredQueries, false)?.length
   const isFiltered = flatQueries?.length !== filteredQueriesCount
+  const tree = useRef<TreeApi<any> | null>(null)
+
+  const query = useSelector(Current.getQuery)
+
+  useEffect(() => {
+    if (!query?.id) {
+      tree.current.select(null, false, false)
+    } else if (!tree.current.getSelectedIds().includes(query.id)) {
+      tree.current.scrollToId(query.id)
+      tree.current.selectById(query.id)
+      setTimeout(() => {
```

Comment:
Oh I forgot, that's the whole reason I updated @types/react-dom. React now "batches" all state updates. I needed the setTimeout to say, "React, please render what you have to the dom so that I can select the item in the tree." However, the selectById code does what I need it to do already. So this setTimeout needs to go.